### DISCUSSION TITLE
Beta: capytaine GUI including meshing tools for windows

### DISCUSSION THREAD

**Discussion by RubendeBruin** [capytaine/capytaine#372] [General]
I've added capytaine to pymeshup and build the whole thing to a stand-alone executable using pyinstaller.

The result is a very simple GUI to make 1st order hydrodynamic databases using capytaine.
The results are saved as .nc and .dhyd using mafredo.

The compiled, stand-alone package can be downloaded from: https://www.open-ocean.org/pymeshup/
The source-code is here: https://github.com/RubendeBruin/pymeshup
Teaser: https://www.linkedin.com/feed/update/urn:li:activity:7087012658922565632?updateEntityUrn=urn%3Ali%3Afs_feedUpdate%3A%28V2%2Curn%3Ali%3Aactivity%3A7087012658922565632%29
Tutorial: https://youtu.be/7SO1BFD3N9A

<img width="863" alt="image" src="https://github.com/capytaine/capytaine/assets/34062862/39d55e19-a822-4207-a67a-cb1c1fdbc022">

Known issues:
- Plotting fails if only a single heading is present

**Comment by mancellin** [comment]
Nice, thank you for working on this!

I downloaded it and gave it a quick try. It seems to work, although I got an error message. I will open an issue in your repository.

**Reply by RubendeBruin** [reply]
Thanks @mancellin  for reviewing. I implemented your comments and pushed to master.

I also added an importer for GHS geometry files.

Unfortunately the .exe created by pyinstaller does not want to work at the moment. It gives me a
```
Traceback (most recent call last):
  File "C:\python\venvs\build\Lib\site-packages\PyInstaller\hooks\rthooks\pyi_rth_pkgutil.py", line 90, in <module>
    _pyi_rthook()
  File "C:\python\venvs\build\Lib\site-packages\PyInstaller\hooks\rthooks\pyi_rth_pkgutil.py", line 31, in _pyi_rthook
    from pyimod02_importers import PyiFrozenImporter
ImportError: cannot import name 'PyiFrozenImporter' from 'pyimod02_importers' (PyInstaller\loader\pyimod02_importers.py)
[23440] Failed to execute script 'pyi_rth_pkgutil' due to unhandled exception!
```
error.

No idea what causes it. Will look into it later.

**Comment by RubendeBruin** [comment]
Ok,
after a reboot it did work.

Updated GUI can be downloaded from: [https://www.open-ocean.org/pymeshup/](https://www.open-ocean.org/pymeshup/)

**Comment by RubendeBruin** [comment]
And there is a quick tutorial here: https://youtu.be/7SO1BFD3N9A
(which results in https://www.linkedin.com/feed/update/urn:li:activity:7087012658922565632?updateEntityUrn=urn%3Ali%3Afs_feedUpdate%3A%28V2%2Curn%3Ali%3Aactivity%3A7087012658922565632%29)
